pkg/plugin/util/util.go

```diff
@@ -117,6 +120,43 @@ func SelfLinkToCRDName(selfLink string) string {
 	}
 }
 
+/*
+ * Extracts the CRD name from a K8S Unstructured into the CRD name.  Supports namespaced and cluster level CRs
+ * K8S Cluster Resource Self Link format: /api/<version>/<resource plural name>/<item name>,
+ *     e.g. /api/v1/persistentvolumes/pvc-3240e5ed-9a97-446c-a6ab-b2442d852d04
+ * K8S Resource Name = <resource plural name>, e.g. persistentvolumes
+ * Custom Resource Cluster Self Link format: /apis/<CR group>/<version>/<CR plural name>/<item name>,
+ *     e.g. /api/cnsdp.vmware.com/v1/backuprepositories/br-1
+ * Custom Resource Name = <CR plural name>.<CR group>, e.g. backuprepositories.cnsdp.vmware.com
+ */
+func UnstructuredToCRDName(item runtime.Unstructured) (string, error) {
+	// Retrieve the common object metadata and check to see if this is a blocked type
+	accessor := meta.NewAccessor()
+	apiVersion, err := accessor.APIVersion(item)
+	if err != nil {
+		return "", err
+	}
+	var group string
+
+	gv, err := schema.ParseGroupVersion(apiVersion)
+	kind, err := accessor.Kind(item)
+	if err != nil {
+		return "", err
+	}
+	group = gv.Group
+
+	var pluralName string
+	if strings.HasSuffix(kind, "y") {
```

Comment:
Can you add the comments you added in constant.go here as well so it is clear why this conversion is done?